auth/auth/driver/driver.py

```diff
@@ -28,6 +28,39 @@
 
 is_test_deployment = DEFAULT_NAMESPACE != 'default'
 
+USER_TIMEOUT_DAYS = os.environ.get("INACTIVE_USER_TIMEOUT_DAYS", "60")
+try:
+    USER_TIMEOUT_DAYS = int(USER_TIMEOUT_DAYS)
+except Exception as exc:
+    raise ValueError("Unable to interpret INACTIVE_USER_TIMEOUT_DAYS as an integer.") from exc
+
+
+async def update_inactive_users(db: Database, user_timeout_days: int):
+    """
+    Deactivates users who have been inactive for longer than the specified timeout.
+
+    Function updates the 'users' table in the database, setting the state to 'inactive'
+    for users who:
+        - are currently 'active';
+        - have a non-null 'last_activated' timestamp;
+        - have been inactive for more than 'user_timeout_days' days; and
+        - are NOT service accounts.
+    Args:
+        db (Database): The database connection object.
+        user_timeout_days (int): The number of days of inactivity after which a user is considered inactive.
+
+    Returns:
+        None
+    """
+    await db.execute_update(
+        """
+UPDATE users
+SET users.state = 'inactive'
+WHERE (users.state = 'active') AND (users.last_activated IS NOT NULL) AND (DATEDIFF(NOW(), users.last_activated) > %s) AND (users.is_service_account = 0);
```

Comment:
Might be worth checking this... but `%s` is usually used for strings, there might be an alternative for numbers that does the interpolation as a number value